engine/src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -256,20 +255,30 @@ class WorkflowActor(workflowId: WorkflowId,
 
   onTransition {
     case oldState -> terminalState if terminalState.terminal =>
-      log.info(s"$tag transition from $oldState to $terminalState: shutting down")
+      workflowLogger.info(s"transition from $oldState to $terminalState: shutting down")
       // Add the end time of the workflow in the MetadataService
       val now = OffsetDateTime.now
-      val metadataEventMsg = MetadataEvent(MetadataKey(workflowId, None, WorkflowMetadataKeys.EndTime),
-        MetadataValue(now))
+      val metadataEventMsg = MetadataEvent(MetadataKey(workflowId, None, WorkflowMetadataKeys.EndTime), MetadataValue(now))
       serviceRegistryActor ! PutMetadataAction(metadataEventMsg)
       terminalState match {
         case WorkflowSucceededState =>
           context.parent ! WorkflowSucceededResponse(workflowId)
         case WorkflowFailedState => // The failure will already have been sent. So just kick back and relax here.
         case WorkflowAbortedState =>
           context.parent ! WorkflowAbortedResponse(workflowId)
-        case unknownState => log.warning(s"$tag $unknownState is an unhandled terminal state!")
+        case unknownState => workflowLogger.warn(s"$unknownState is an unhandled terminal state!")
       }
+
+      // Copy/Delete workflow logs
+      if (WorkflowLogger.isEnabled) {
+        stateData.workflowDescriptor foreach { wd =>
+          // Create the actor in the system context because this WorkflowActor is going to die right after
+          val copyWorkflowLogsActor: ActorRef = context.system.actorOf(CopyWorkflowLogsActor.props(wd), s"CopyWorkflowLogsActor-$workflowId")
+          copyWorkflowLogsActor ! CopyWorkflowLogsActor.Start
```

Comment:
I would love to see more "let it crash" behaviours with proper supervision strategy etc... It could actually save us up some work and more importantly a lot of messages going back and forth. I honestly felt like this being the "last" ticket before 0.20, spending time doing a perfect let it crash model for this particular feature wasn't worth doing now. But I'm happy to create a ticket for it or even make a subsequent PR later.